### PR TITLE
Added version patterns to user-guide.rst

### LINKED ISSUE #1604: Document version pattern for calendar versioning
It seems that `version pattern` field documentation is missing in user guide. It would be good to add it, so people know how to fill this field correctly.

### PR DESCRIPTION
**Pull request by vidit-maheshwari**
### Issue

Fixes #1604 

### Description

This pull request adds documentation for the version pattern field in the user guide. The version pattern field is essential for extracting and matching version numbers from project releases, and it allows users to specify the format of version numbers using regular expressions. This documentation clarifies the usage of various pattern tokens such as `YYYY`, `YY`, `MM`, `WW`, `DD`, etc., along with their examples and notes on usage.

### Changes Made

- Added a new section titled "Version Pattern" to the user guide (`user-guide.rst`).
- Provided detailed explanations and examples for each pattern token (`YYYY`, `YY`, `MM`, `WW`, `DD`, etc.).
- Included notes on traditional version numbering, the relative nature of short and zero-padded years, and the mutual exclusivity of weeks with months/days.

### Testing Done

- Reviewed the changes locally to ensure accuracy and clarity.
- Checked for formatting and consistency with existing documentation.

### Additional Notes

- This documentation enhancement aims to improve usability and clarity for users configuring the version pattern field.
- Any feedback or suggestions for improvement are welcome.

---
*Please review and merge this pull request at your earliest convenience. Thank you!*

### CODE REVIEW

**Comment by vidit-maheshwari** [comment]
@Zlopez Everything looks fine now and i have also ran all the tests locally before making the PR and everything looks clean.
Kindly review this PR

File: docs/user-guide.rst [hunk 1]
```diff
@@ -296,6 +296,31 @@ Anitya provides several different versions schemes.
     recent scheme and the rest will be moved to bottom unsorted.
 
 
+Version Pattern
+===============
+
+The version pattern is used to extract and match version numbers from
+project releases. It allows for specifying the format of version numbers
```
Comment (Zlopez) [inline docs/user-guide.rst:303]:
```suggestion
project releases. It allows for specifying the format for calendar versioning scheme using following patterns.
```

File: docs/user-guide.rst [hunk 2]
```diff
@@ -296,6 +296,31 @@ Anitya provides several different versions schemes.
...
+using regular expressions.
```
Comment (Zlopez) [inline docs/user-guide.rst:304]:
```suggestion
```

File: docs/user-guide.rst [hunk 3]
```diff
@@ -296,6 +296,31 @@ Anitya provides several different versions schemes.
...
+
+Patterns -
```
Comment (Zlopez) [inline docs/user-guide.rst:306]: You should add empty line after this, otherwise the list will not work correctly.

File: docs/user-guide.rst [hunk 4]
```diff
@@ -296,6 +296,31 @@ Anitya provides several different versions schemes.
...
+- ``YY``: Short year (e.g., 6, 16, 106).
+- ``0Y``: Zero-padded year (e.g., 06, 16, 106).
+- ``MM``: Short month (e.g., 1, 2, ..., 11, 12).
+- ``0M``: Zero-padded month (e.g., 01, 02, ..., 11, 12).
+- ``WW``: Short week (since the start of the year) (e.g., 1, 2, 33, 52).
+- ``0W``: Zero-padded week (e.g., 01, 02, 33, 52).
+- ``DD``: Short day (e.g., 1, 2, ..., 30, 31).
+- ``0D``: Zero-padded day (e.g., 01, 02, ..., 30, 31).
+
+Notes:
```
Comment (Zlopez) [inline docs/user-guide.rst:317]: Use notes from restructured text https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html#notes

File: docs/user-guide.rst [hunk 5]
```diff
@@ -296,6 +296,31 @@ Anitya provides several different versions schemes.
     recent scheme and the rest will be moved to bottom unsorted.
 
 
+Version Pattern
+===============
+
+The version pattern is used to extract and match version numbers from
```
Comment (Zlopez) [inline docs/user-guide.rst:302]:
```suggestion
The version pattern is used to define how the calendar version should be parsed and how they will be sorted. 
```

**Review by Zlopez — CHANGES_REQUESTED** [review]
Needs few changes.

**Comment by vidit-maheshwari** [comment]
@Zlopez i have made the required changes as you instructed. Kindly review if needed any changes

**Comment by Zlopez** [comment]
The documentation seems OK now, but you introduced error to docs CI, could you fix that?

**Comment by vidit-maheshwari** [comment]
@Zlopez I have resolved the build issue, Kindly review the PR.

**Comment by Zlopez** [comment]
@vidit-maheshwari Thanks for resolving the failing tests. Could you just squash the commits? We don't need 6 commits for one documentation change :-)

**Pull request closed without merging**